OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs

```diff
@@ -194,17 +197,15 @@ void Deploy(bool init)
 
 			if (!string.IsNullOrEmpty(info.DeploySound))
 				Game.Sound.Play(info.DeploySound, self.CenterPosition);
-
+			if (!string.IsNullOrEmpty(info.DeployAnimation))
+				body.Value.PlayCustomAnimation(self, info.DeployAnimation);
```

Comment:
This will break the Mobile Sensor Array/Tick Tank/the rest of the deploying TS vehicles.

Their `SpriteBody` is enabled by the upgrades granted below, because they are voxel actors while mobile, and sprites while deployed. That's why the upgrades have to come first.